pyblox3/src/abtesting.py

```diff
@@ -6,16 +6,75 @@
+# code for it, my head hurts.
+
+# TODO Consider removing my existence
+# FIXME The awful code below.
+
 from .util import *
-import json
+
+root = "https://abtesting.roblox.com"
+
+
+# TODO Make this less verbose and use kwargs
+def createGetEnrollmentType(ExperimentName: str, SubjectType: str or int, SubjectTargetId: int) -> dict:
+    """
+    OK, SubjectType can only be 2, 1 throws a 403 (I assume it requires some authorization)
+    -> when looking into it more, it appears that nsg has no clue what SubjectType 1 means and/or how to use it ;)
+    -> OK nsg, you definitely didn't look into it more, SubjectType 1 is user, and requires a valid userId,
+        basically the one of the currently authenticated user:
+        1 is User
+    """
+    if len(ExperimentName) == 0:
+        raise SyntaxError("The ExperimentName shouldn't be empty.")
+    if len(ExperimentName) > 50:
+        raise SyntaxError("The ExperimentName shouldn't be longer than 50 characters.")
+    if SubjectTargetId < 1:
+        raise SyntaxError("In order to satisfy the request, the SubjectTargetId is required to be greater than 0.")
+    if (type(SubjectType) == int and SubjectType > 2) \
+            or (type(SubjectType) == int and SubjectType < 1) \
+            or (type(SubjectType) == str and len(SubjectType) == 0):
+        raise SyntaxError("In order to satisfy the request, the SubjectType is required to 1 or 2 "
+                          "or User or BrowserTracker")
+    return {'ExperimentName': ExperimentName, 'SubjectType': SubjectType, 'SubjectTargetId': SubjectTargetId}
+
 
 class AbTesting_v1:
 
-    def __init__(self,**kwargs):
+    def __init__(self, **kwargs):
         groupid = kwargs.get("groupid", None)
         auth = kwargs.get("auth", None)
         self.groupid = groupid
         self.auth = auth.auth_cookies
-    
-    def __str__(self):
-        return "AbTesting_v1 class doesn't have any endpoints"
\ No newline at end of file
+
+    # POST https://abtesting.roblox.com/v1/get-enrollments
+    async def getEnrollments(self,
+                             **kwargs):
+        """
+        OK, it should consist of an array of these:
+        {
+            "ExperimentName": "SomeABTest",
+            "SubjectType": int,
+            "SubjectTargetId": int64,
+        }
+        you should use createGetEnrollmentType as a helper
+        """
+        global root
+        auth = self.auth
+        request = kwargs.get('request')
+        response = await Req.request(t='POST',
+                                     url=root+'/v1/get-enrollments',
+                                     cookies=auth,
+                                     payload=request,
+                                     header={'Content-Type': 'application/json'})
+        if response[0] == 200:
+            return response[4]['data']
+        elif response[0] == 403:
+            raise RuntimeError("You are unauthorised to create this request with the credentials you supplied!")
```

Comment:
Error codes are handled automatically by the library. You could just return ``return False, response[4]``. POST, PATCH, DEL are treated as tuples to allow for comparison and an easy what to "prove" confirmation that the action completed (true) or not (false).